enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,588 @@
+installer's current REST API.
+
+The details of this are being discussed on [a hive
+pull request](https://github.com/openshift/hive/pull/1247).
+
+
+**Agent**
+Agent is a new resource, part of the assisted installer's new operator, that
+represents a host that is destined to become part of an OpenShift cluster, and
+is running an agent that is able to run inspection and installation tasks. It
+correlates to the "host" resource in the Assisted Installer's REST API.
+
+The details of this API are being discussed in a [pull
+request](https://github.com/openshift/assisted-service/pull/861) that
+implements the CRD.
+
+
+#### REST API Access
+
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
+authentication headers, the Assisted Service must generate some random URL or
+query parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+Agent CRs itself, the agent will continue communicating with the service via
+REST APIs. Currently the service embeds the user’s pull secret in the discovery
+ISO which the agent passes in an authentication header.  In this case the
+service can generate and embed some token which it can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include a new `AgentBareMetal` platform that
+contains such fields as:
+
+* API VIP
+* API VIP DNS name
+* IngressVIP
+* VIPDHCPAllocation
+
+A new `InstallStrategy` section of the Spec enables the API to describe a way
+of installing a cluster other than the default of using `openshift-install`.
+The new field has an "agent" option that includes such fields as:
+
+* AgentSelector, a label selector for identifying which Agent resources should
+be included in the cluster.
+* ProvisionRequirements, where the API user can specify how many agents to
+expect before beginning installation for each of the control-plane and worker
+roles.
+
+#### Use of metal3
+
+[metal3](https://metal3.io/) will be used to boot the assisted discovery ISO
+for bare-metal deployments. Specifically, the BareMetalHost API will be used
+from a hub cluster to boot the discovery ISO on hosts that will be used to
+form new clusters.
+
+For this first iteration, it is assumed that the hub cluster itself is using
+the bare metal platform, and thus will have baremetal-operator installed
+and available for use. In the future it may be desirable to make the metal3
+capabilities available on hub clusters that are not using the bare metal
+platform.
```

Comment:
so what platform would the hub clusters be using?
I assume baremetal, just like currently with no bmcDetails (so that's why ironic can't be used on the hub)?

Reply:
The platform of the hub cluster is baremetal, and the scenario that this document focuses on is leveraging metal3 and ironic to boot the Assisted discovery ISO using virtual media.  The BMHs representing hosts of the spoke cluster will have bmcDetails.